UI/src/view/point_asset/trafficSignForm.js

```diff
@@ -539,11 +385,19 @@
       return '<button class="btn edit-only editable btn-secondary add-panel" id="'+id+'" >Uusi lisäkilpi</button>';
     };
 
-    var singleChoiceForPanelTypes = function (property, collection) {
+    var singleChoiceForPanelTypes = function (property, collection, asset) {
       var propertyValue = _.isUndefined(_.last(property))  ? '' : _.last(property);
       var signTypes = _.map( getValuesFromEnumeratedProperty('trafficSigns_type') );
       var panels = _.find(collection.getAdditionalPanels(signTypes));
-      var propertyDisplayValue = _.find(panels, function(panel){return panel.propertyValue == propertyValue.toString();}).propertyDisplayValue;
+      var propertyDisplayValue;
+
+      if (isPedestrianOrCyclingRoadLink(asset)) {
+        panels = _.filter(panels, function(p) { if (p.propertyValue == '61') return p;} );
+        propertyDisplayValue = _.isUndefined(panels) && panels.length === 0 ? "" : panels[0].propertyDisplayValue;
```

Comment:
Short circuit of &&
IF panels is undefined, the IF will not do the validation of the length.

So if panels exists and the have 1 or more...will get information about panels[0]

Reply:
This looks fine to me. As @ricardofmsimoes said, if the first condition fails it won't reach the second, if it passes then panels is not undefined

Reply:
_.isUndefined(value) = Returns true if value is undefined, else false.


So imagine the panels are undefined(_.isUndefined(panels) = true) then it will check the second condition and then give an error because undefined.length is impossible since undefined does not have a length.

Now imagine that is not undefined, then will return false and not check the second condition since the first one is false.


That's why I think this '&&' condition will always give an error or false and never true but maybe I'm confusing with another thing.

Reply:
Code was altered as suggested